### DISCUSSION TITLE
Have you considered leveraging `elpaca`?

### DISCUSSION THREAD

**Discussion by panchoh** [jamescherti/minimal-emacs.d#15] [Ideas]
Hi!

Doom Emacs user here, currently building my .emacs config from scratch (if only to eventually be able to help more on the Doom Emacs project).

Thanks for this config! Really useful to be able to see `use-package` used to configure everything so cleanly.

So, have you considered leveraging progfolio’s [Elpaca](https://github.com/progfolio/elpaca) to further improve the installation and maintenance of the packages?

I’d recommend this [video](https://youtu.be/5Ud-TE3iIQY) by the author to see it in action. He maintained `straight.el`, and eventually produced a better package management system that tackles some of its limitations. Ah, Doom Emacs will most likely switch to it.

Just a thought.

Thanks again!!!

**Comment by jamescherti** [comment]
Hi @panchoh,

I'm glad you found the `minimal-emacs.d` configuration useful!

Building your own `.emacs.d` config from scratch will definitely give you more insight into Emacs itself, and your contributions to Doom will be all the stronger for it.

The philosophy behind `minimal-emacs.d` is to remain as minimal as possible, leaving users the choice to activate only the modes they find relevant. Users can add Elpaca to their minimal-emacs.d configuration by [adding the Elpaca bootstrap code](https://github.com/progfolio/elpaca?tab=readme-ov-file#installer) to their `~/.emacs.d/pre-init.el`.

Thanks for the suggestion. I’ve just added Elpaca to the `README.md` file.

**Reply by panchoh** [reply]
Thanks for adding the instructions to setup Elpaca!

Alas, one small but important detail seems to be missing, namely, the addition of `(setq package-enable-at-startup nil)` to `early-init.el`, as described at the end of the [section](https://github.com/progfolio/elpaca?tab=readme-ov-file#installer), just before the [Quickstart](https://github.com/progfolio/elpaca?tab=readme-ov-file#quick-start).

(Or I can do a PR if preferred).

Thanks again!

**Reply by jamescherti** [reply]
Thank you for pointing that out, @panchoh! I have added `(setq package-enable-at-startup nil)` to the README.md file.

**Reply by panchoh** [reply]
Very much appreciated!
